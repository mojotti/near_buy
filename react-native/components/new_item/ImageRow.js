import React from 'react';
import { Alert, View } from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import PropTypes from 'prop-types';
import { styles } from '../../src/static/styles/NewItemStyles';

export class ImageRow extends React.Component {
  render() {
    return (
      <View style={styles.imageRow}>
        <MaterialIcons.Button
          name={'add-a-photo'}
          backgroundColor={'transparent'}
          size={100}
          color={'gray'}
          style={styles.addPhotoButton}
          onPress={() => Alert.alert('You pressed me')}
        />
        <MaterialIcons.Button
          name={'add-a-photo'}
          backgroundColor={'transparent'}
          size={100}
          color={'gray'}
          style={styles.addPhotoButton}
          onPress={() => Alert.alert('You pressed me')}
        />
      </View>
    );
  }
}

ImageRow.propTypes = {
  onLeftImageSelected: PropTypes.func.isRequired,
  onRightImageSelected: PropTypes.func.isRequired,
};
