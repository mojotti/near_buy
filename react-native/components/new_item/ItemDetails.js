import React from 'react';
import { TextInput, View } from 'react-native';
import PropTypes from 'prop-types';
import { styles } from '../../src/static/styles/NewItemStyles';

export class ItemDetails extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      price: '',
      description: '',
    };

    this.handleTitleChange = this.handleTitleChange.bind(this);
    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
    this.handlePriceChange = this.handlePriceChange.bind(this);
  }

  handleTitleChange(text) {
    this.setState({ title: text });
    this.props.onTitleChange(text);
  }

  handleDescriptionChange(text) {
    this.setState({ description: text });
    this.props.onDescriptionChange(text);
  }

  handlePriceChange(text) {
    let newText = '';
    const numbers = '0123456789';

    for (let i = 0; i < text.length; i++) {
      if (numbers.indexOf(text[i]) > -1) {
        newText += text[i];
      }
    }

    this.setState({ price: newText });
    this.props.onPriceChange(this.state.price);
  }

  render() {
    return (
      <View>
        <View style={styles.itemDetailContainer}>
          <TextInput
            placeholder="Title"
            autoCapitalize="sentences"
            autoCorrect={false}
            autoFocus={this.state.mounted}
            maxLength={40}
            keyboardType="email-address"
            value={this.state.title}
            onChangeText={text => this.handleTitleChange(text)}
            style={[styles.itemDetails]}
            underlineColorAndroid="transparent"
          />
        </View>
        <View style={styles.itemDescriptionContainer}>
          <TextInput
            placeholder="Description"
            autoCapitalize="sentences"
            maxLength={400}
            autoCorrect={false}
            keyboardType="email-address"
            value={this.state.description}
            multiline={true}
            onChangeText={text => this.handleDescriptionChange(text)}
            style={[styles.itemDetails]}
            underlineColorAndroid="transparent"
          />
        </View>
        <View style={styles.itemDetailContainer}>
          <TextInput
            placeholder="Price"
            autoCorrect={false}
            autoFocus={false}
            keyboardType="numeric"
            maxLength={5}
            value={this.state.price}
            onChangeText={text => this.handlePriceChange(text)}
            style={[styles.itemDetails]}
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    );
  }
}

ItemDetails.propTypes = {
  onTitleChange: PropTypes.func.isRequired,
  onDescriptionChange: PropTypes.func.isRequired,
  onPriceChange: PropTypes.func.isRequired,
};
