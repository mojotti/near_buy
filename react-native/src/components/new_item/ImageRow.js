import React from 'react';
import { Alert, Image, TouchableHighlight, View } from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import PropTypes from 'prop-types';
import ImagePicker from 'react-native-image-crop-picker';
import { cameraIconSize, styles } from '../../static/styles/NewItemStyles';

export class ImageRow extends React.Component {
  constructor(props) {
    super(props);
    this.handleImageSelection = this.handleImageSelection.bind(this);
    this.handleNewImage = this.handleNewImage.bind(this);
    this.handleGalleryImage = this.handleGalleryImage.bind(this);
    this.renderEmptyImageButton = this.renderEmptyImageButton.bind(this);
    this.renderImageButton = this.renderImageButton.bind(this);
  }

  handleImageSelection(imageId) {
    Alert.alert(
      'Select picture',
      'Select from gallery or take a new one with camera',
      [
        {
          text: 'Cancel',
          onPress: () => {},
          style: 'cancel',
        },
        {
          text: 'Gallery',
          onPress: () => this.handleGalleryImage(imageId),
        },
        { text: 'Camera', onPress: () => this.handleNewImage(imageId) },
      ],
      { cancelable: false },
    );
  }

  handleNewImage(imageId) {
    ImagePicker.openCamera({
      width: 400,
      height: 400,
      cropping: true,
    })
      .then(image => this.props.onImageSelected(image, imageId))
      .catch(error => console.log('error in image selection:', error));
  }

  handleGalleryImage(imageId) {
    ImagePicker.openPicker({
      width: 400,
      height: 400,
      cropping: true,
    })
      .then(image => this.props.onImageSelected(image, imageId))
      .catch(error => console.log('error in image selection:', error));
  }

  renderEmptyImageButton(buttonId) {
    return (
      <MaterialIcons.Button
        name={'add-a-photo'}
        backgroundColor={'transparent'}
        size={cameraIconSize}
        color={'gray'}
        style={styles.addPhotoButton}
        onPress={() => this.handleImageSelection(buttonId)}
      />
    );
  }

  renderImageButton(buttonId) {
    const index = buttonId % 2 === 0 ? 0 : 1;
    return (
      <TouchableHighlight onPress={() => this.handleImageSelection(buttonId)}>
        <Image
          style={styles.photoStyles}
          source={{ uri: this.props.images[index].path }}
        />
      </TouchableHighlight>
    );
  }

  render() {
    return (
      <View style={styles.imageRow}>
        {this.props.images[0] === null
          ? this.renderEmptyImageButton(this.props.leftButtonId)
          : this.renderImageButton(this.props.leftButtonId)}

        {this.props.images[1] === null
          ? this.renderEmptyImageButton(this.props.rightButtonId)
          : this.renderImageButton(this.props.rightButtonId)}
      </View>
    );
  }
}

ImageRow.propTypes = {
  onImageSelected: PropTypes.func.isRequired,
  leftButtonId: PropTypes.number.isRequired,
  rightButtonId: PropTypes.number.isRequired,
  images: PropTypes.arrayOf(
    PropTypes.shape({
      size: PropTypes.number,
      mime: PropTypes.string,
      height: PropTypes.number,
      width: PropTypes.number,
      modificationDate: PropTypes.string,
      path: PropTypes.string,
      cropRect: PropTypes.shape({
        height: PropTypes.number,
        width: PropTypes.number,
        y: PropTypes.number,
        x: PropTypes.number,
      }),
    }),
  ).isRequired,
};
