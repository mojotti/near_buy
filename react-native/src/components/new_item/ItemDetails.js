import React from 'react';
import { TextInput, View } from 'react-native';
import PropTypes from 'prop-types';
import { styles } from '../../static/styles/NewItemStyles';

export class ItemDetails extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      price: '',
      description: '',
    };

    this.handleTitleChange = this.handleTitleChange.bind(this);
    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
    this.handlePriceChange = this.handlePriceChange.bind(this);
  }

  componentDidMount() {
    if (this.props.title && this.props.description && this.props.price) {
      this.setState(() => ({
        title: this.props.title,
        description: this.props.description,
        price: this.props.price.toString(),
      }));
    }
  }

  handleTitleChange(text) {
    this.setState({ title: text });
    this.props.onTitleChange(text);
  }

  handleDescriptionChange(text) {
    this.setState({ description: text });
    this.props.onDescriptionChange(text);
  }

  handlePriceChange(text) {
    let newText = '';
    const numbers = '0123456789';

    for (let i = 0; i < text.length; i++) {
      if (numbers.indexOf(text[i]) > -1) {
        newText += text[i];
      }
    }

    this.setState({ price: newText });
    this.props.onPriceChange(newText);
  }

  render() {
    return (
      <View>
        <View style={styles.itemDetailContainer}>
          <TextInput
            placeholder="Title"
            autoCapitalize="sentences"
            autoCorrect={false}
            maxLength={40}
            keyboardType="email-address"
            value={this.state.title}
            onChangeText={text => this.handleTitleChange(text)}
            style={[styles.itemDetails]}
            underlineColorAndroid="transparent"
          />
        </View>
        <View style={styles.itemDescriptionContainer}>
          <TextInput
            placeholder="Description"
            autoCapitalize="sentences"
            maxLength={400}
            autoCorrect={false}
            keyboardType="email-address"
            value={this.state.description}
            multiline={true}
            onChangeText={text => this.handleDescriptionChange(text)}
            style={[styles.itemDetails]}
            underlineColorAndroid="transparent"
          />
        </View>
        <View style={styles.itemDetailContainer}>
          <TextInput
            placeholder="Price"
            autoCorrect={false}
            autoFocus={false}
            keyboardType="numeric"
            maxLength={5}
            value={this.state.price}
            onChangeText={text => this.handlePriceChange(text)}
            style={[styles.itemDetails]}
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    );
  }
}

ItemDetails.defaultProps = {
  title: '',
  description: '',
  price: null,
};

ItemDetails.propTypes = {
  onTitleChange: PropTypes.func.isRequired,
  onDescriptionChange: PropTypes.func.isRequired,
  onPriceChange: PropTypes.func.isRequired,
  title: PropTypes.string,
  description: PropTypes.string,
  price: PropTypes.number,
};
