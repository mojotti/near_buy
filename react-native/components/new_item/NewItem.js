'use strict';

import React from 'react';
import store from '../../redux/index';
import {
  Alert,
  ScrollView,
  Text,
  TouchableHighlight,
  View,
} from 'react-native';
import { NavigationActions } from 'react-navigation';
import {
  alertInvalidValuesNewItem,
  localhost,
} from '../../src/static/constants';
import { styles } from '../../src/static/styles/NewItemStyles';
import { ItemDetails } from './ItemDetails';
import { ImageRow } from './ImageRow';

const UPPER_BUTTON_IDS = {
  leftButtonId: 0,
  rightButtonId: 1,
};

const LOWER_BUTTON_IDS = {
  leftButtonId: 2,
  rightButtonId: 3,
};

export class NewItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      price: '',
      description: '',
      mounted: false,
      images: [null, null, null, null],
    };

    this.handleTitleChange = this.handleTitleChange.bind(this);
    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
    this.handlePriceChange = this.handlePriceChange.bind(this);
    this.setNewImage = this.setNewImage.bind(this);
  }

  static navigationOptions = {
    title: 'Add item',
  };

  componentDidMount() {
    // work around for testing issue, should be solved when RN version is updated
    this.setState({ mounted: true });
  }

  getHeaders() {
    const headers = new Headers();
    headers.append('Content-Type', 'application/json');
    headers.append('Authorization', `Bearer ${store.getState().auth.token}`);
    return headers;
  }

  resetNavigationAndNavigateToRoute(targetRoute) {
    const resetAction = NavigationActions.reset({
      index: 0,
      actions: [NavigationActions.navigate({ routeName: targetRoute })],
    });
    this.props.navigation.dispatch(resetAction);
  }

  handleNewItemCreation() {
    const { title, price, description } = this.state;
    if (title === '' || price === '' || description === '') {
      Alert.alert(...alertInvalidValuesNewItem);
      return;
    }
    const url = `http://${localhost}:5000/api/v1.0/items`;
    fetch(url, {
      method: 'POST',
      body: JSON.stringify({
        title: title,
        price: price,
        description: description,
      }),
      headers: this.getHeaders(),
    })
      .then(response => response.json())
      .then(responseJson => {
        if (responseJson.item && responseJson.item.title === title) {
          this.resetNavigationAndNavigateToRoute('Items');
        } else {
          Alert.alert('Item creation failed', 'Something went wrong');
        }
      })
      .catch(error => {
        console.error(error);
      });
  }

  renderSubmitButton() {
    return (
      <TouchableHighlight
        id="SubmitButton"
        onPress={() => this.handleNewItemCreation()}
        style={styles.submitButton}
      >
        <Text style={styles.submitText}>{'Submit item'}</Text>
      </TouchableHighlight>
    );
  }

  setNewImage(image, id) {
    this.setState(prevState => {
      let images = prevState.images;
      images[id] = image;
      return images;
    });
    console.log('setting new image: ', image, id);
    console.log(this.state.images);
  }

  handleTitleChange(text) {
    this.setState({ title: text });
  }

  handleDescriptionChange(text) {
    this.setState({ description: text });
  }

  handlePriceChange(text) {
    this.setState({ price: text });
  }

  render() {
    const itemDetailsProps = {
      onTitleChange: this.handleTitleChange,
      onDescriptionChange: this.handleDescriptionChange,
      onPriceChange: this.handlePriceChange,
    };

    const firstImageRowProps = {
      onImageSelected: this.setNewImage,
      ...UPPER_BUTTON_IDS,
      images: this.state.images.slice(0, 2),
    };

    const secondImageRowProps = {
      onImageSelected: this.setNewImage,
      ...LOWER_BUTTON_IDS,
      images: this.state.images.slice(2),
    };

    return (
      <ScrollView>
        <View style={styles.container}>
          <Text style={styles.headerText}>Item details</Text>
          <ItemDetails {...itemDetailsProps} />
          <Text style={styles.headerText}>Add pictures</Text>
          <ImageRow {...firstImageRowProps} style={styles.imageRowMargin} />
          <ImageRow {...secondImageRowProps} style={styles.imageRowMargin} />
          {this.renderSubmitButton()}
        </View>
      </ScrollView>
    );
  }
}

export default NewItem;
