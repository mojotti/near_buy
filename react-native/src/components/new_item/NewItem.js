'use strict';

import React from 'react';
import store from '../../../redux/index';
import {
  Alert,
  ScrollView,
  Text,
  TouchableHighlight,
  View,
} from 'react-native';
import { NavigationActions } from 'react-navigation';
import {
  alertInvalidValuesNewItem,
  localhost,
} from '../../static/constants';
import { styles } from '../../static/styles/NewItemStyles';
import { ItemDetails } from './ItemDetails';
import { ImageRow } from './ImageRow';

const UPPER_BUTTON_IDS = {
  leftButtonId: 0,
  rightButtonId: 1,
};

const LOWER_BUTTON_IDS = {
  leftButtonId: 2,
  rightButtonId: 3,
};

export class NewItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      price: '',
      description: '',
      images: [null, null, null, null],
    };

    this.handleTitleChange = this.handleTitleChange.bind(this);
    this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
    this.handlePriceChange = this.handlePriceChange.bind(this);
    this.setNewImage = this.setNewImage.bind(this);
  }

  static navigationOptions = {
    title: 'Add item',
  };

  getHeaders() {
    const headers = new Headers();
    headers.append('Content-Type', 'application/json');
    headers.append(
      'Authorization',
      `Bearer ${store.getState().authorizationReducer.token}`,
    );
    return headers;
  }

  resetNavigationAndNavigateToRoute(targetRoute) {
    const resetAction = NavigationActions.reset({
      index: 0,
      actions: [NavigationActions.navigate({ routeName: targetRoute })],
    });
    this.props.navigation.dispatch(resetAction);
  }

  handleNewItemCreation() {
    const { title, price, description } = this.state;
    if (title === '' || price === '' || description === '') {
      Alert.alert(...alertInvalidValuesNewItem);
      return;
    }
    const url = `http://${localhost}:5000/api/v1.0/items`;
    fetch(url, {
      method: 'POST',
      body: JSON.stringify({
        title: title,
        price: price,
        description: description,
      }),
      headers: this.getHeaders(),
    })
      .then(response => response.json())
      .then(responseJson => {
        this.handleResponse(responseJson);
      })
      .catch(error => {
        console.error(error);
      });
  }

  handleResponse(responseJson) {
    if (responseJson.item && responseJson.item.title === this.state.title) {
      this.resetNavigationAndNavigateToRoute('Items');
    } else {
      Alert.alert('Item creation failed', 'Something went wrong');
    }
  }

  renderSubmitButton() {
    return (
      <TouchableHighlight
        id="SubmitButton"
        onPress={() => this.handleNewItemCreation()}
        style={styles.submitButton}
      >
        <Text style={styles.submitText}>{'Submit item'}</Text>
      </TouchableHighlight>
    );
  }

  setNewImage(image, id) {
    this.setState(prevState => {
      let images = prevState.images;
      images[id] = image;
      return images;
    });
  }

  handleTitleChange(text) {
    this.setState({ title: text });
  }

  handleDescriptionChange(text) {
    this.setState({ description: text });
  }

  handlePriceChange(text) {
    this.setState({ price: text });
  }

  render() {
    const itemDetailsProps = {
      onTitleChange: this.handleTitleChange,
      onDescriptionChange: this.handleDescriptionChange,
      onPriceChange: this.handlePriceChange,
    };

    const firstImageRowProps = {
      onImageSelected: this.setNewImage,
      ...UPPER_BUTTON_IDS,
      images: this.state.images.slice(0, 2),
    };

    const secondImageRowProps = {
      onImageSelected: this.setNewImage,
      ...LOWER_BUTTON_IDS,
      images: this.state.images.slice(2),
    };

    return (
      <ScrollView>
        <View style={styles.container}>
          <Text style={styles.headerText}>Item details</Text>
          <ItemDetails {...itemDetailsProps} />
          <Text style={styles.headerText}>Add pictures</Text>
          <ImageRow {...firstImageRowProps} style={styles.imageRowMargin} />
          <ImageRow {...secondImageRowProps} style={styles.imageRowMargin} />
          {this.renderSubmitButton()}
        </View>
      </ScrollView>
    );
  }
}

export default NewItem;
