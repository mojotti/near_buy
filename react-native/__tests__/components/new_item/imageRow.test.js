import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import 'react-native-image-crop-picker';
import { ImageRow } from '../../../components/new_item/ImageRow';

jest.mock('react-native-image-crop-picker', () => {
  return {
    openPicker: jest.fn(() => Promise.resolve()),
    openCamera: jest.fn(() => Promise.resolve()),
  };
});

const PROPS_NO_IMAGES = {
  onImageSelected: () => {},
  leftButtonId: 2,
  rightButtonId: 3,
  images: [null, null, null, null],
};

const PROPS_WITH_IMAGES = {
  onImageSelected: () => {},
  leftButtonId: 0,
  rightButtonId: 1,
  images: [{ path: 'foo/bar.png' }, { path: 'baz/foo.png' }],
};

describe('<ImageRow />', () => {
  test('when there is no images, empty image buttons are rendered', () => {
    const imageRowComponent = shallow(<ImageRow {...PROPS_NO_IMAGES} />);
    expect(imageRowComponent).toMatchSnapshot();
  });

  test('when there are images, images are rendered', () => {
    const imageRowComponent = shallow(<ImageRow {...PROPS_WITH_IMAGES} />);
    expect(imageRowComponent).toMatchSnapshot();
  });

  test('image is selected for empty image', () => {
    const handleImageSelectionSpy = sinon.spy(
      ImageRow.prototype,
      'handleImageSelection',
    );
    const imageRowComponent = shallow(<ImageRow {...PROPS_NO_IMAGES} />);
    imageRowComponent
      .find('IconButton')
      .first()
      .simulate('Press');

    expect(handleImageSelectionSpy.callCount).toEqual(1);
    handleImageSelectionSpy.restore();
  });

  test('image is selected for already selected image', () => {
    const handleImageSelectionSpy = sinon.spy(
      ImageRow.prototype,
      'handleImageSelection',
    );

    const imageRowComponent = shallow(<ImageRow {...PROPS_WITH_IMAGES} />);
    imageRowComponent
      .find('TouchableHighlight')
      .first()
      .simulate('Press');

    expect(handleImageSelectionSpy.callCount).toEqual(1);
  });

  test('onImageSelected is called when selecting camera image', () => {
    const props = {
      onImageSelected: sinon.spy(),
      leftButtonId: 0,
      rightButtonId: 1,
      images: [{ path: 'foo/bar.png' }, { path: 'baz/foo.png' }],
    };

    const imageRowComponent = shallow(<ImageRow {...props} />);
    const imageId = 0;

    imageRowComponent.instance().handleNewImage(imageId);
  });
});
