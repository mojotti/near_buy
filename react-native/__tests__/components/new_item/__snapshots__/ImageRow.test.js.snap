// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageRow /> when there are images, images are rendered 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onPress={[Function]}
    underlayColor="black"
  >
    <Image
      source={
        Object {
          "uri": "foo/bar.png",
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "flex": 0.4,
          "height": 262.5,
          "margin": 10,
          "width": 262.5,
        }
      }
    />
  </TouchableHighlight>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onPress={[Function]}
    underlayColor="black"
  >
    <Image
      source={
        Object {
          "uri": "baz/foo.png",
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "flex": 0.4,
          "height": 262.5,
          "margin": 10,
          "width": 262.5,
        }
      }
    />
  </TouchableHighlight>
</Component>
`;

exports[`<ImageRow /> when there is no images, empty image buttons are rendered 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <IconButton
    backgroundColor="transparent"
    borderRadius={5}
    color="gray"
    name="add-a-photo"
    onPress={[Function]}
    size={210.00000000000003}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 1,
        "flex": 0.4,
        "height": 262.5,
        "justifyContent": "center",
        "margin": 14,
        "width": 262.5,
      }
    }
  />
  <IconButton
    backgroundColor="transparent"
    borderRadius={5}
    color="gray"
    name="add-a-photo"
    onPress={[Function]}
    size={210.00000000000003}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 1,
        "flex": 0.4,
        "height": 262.5,
        "justifyContent": "center",
        "margin": 14,
        "width": 262.5,
      }
    }
  />
</Component>
`;
